packages/plugin-ma-tracking/src/advanced-campaign-tracker.ts

```diff
@@ -0,0 +1,43 @@
+import { Campaign, CampaignTrackerOptions } from '@amplitude/analytics-types';
+import { CampaignTracker } from '@amplitude/analytics-client-common';
+
+export class AdvancedCampaignTracker extends CampaignTracker {
```

Comment:
this class seems to act as a bridge for `ma-tracking` and `CampaignTracker` and comes out a little odd as it tries to control two different versions of a feature. i'd recommend that we write out our own tracker for the plugin and let the old logic live on it's own. this way they're always separate and no changes impact both.

Reply:
In stead of extending the `CampaignTracker`, I initialize and use the `CampaignTracker` instance directly because it contains enough methods I need.